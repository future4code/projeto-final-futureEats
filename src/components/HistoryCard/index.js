import React from 'react';
import styled from 'styled-components';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const StyledCard= styled(Card)`
    margin-top:8px;
    width: 328px;
    height: 102px;
`

const StyledCardContent = styled(CardContent)`
    border-radius:8px;
    border: solid 1px #b8b8b8;
    width: 328px;
    height: 102px;
    display:flex;
    flex-direction:column;
    justify-content: space-around;
    padding:16px ;
`

const StyledTypographyRestaurantTitle = styled(Typography)`
    width: 296px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: #5cb646;
`

const StyledTypographyData = styled(Typography)`
    width: 296px;
    height: 18px;
    font-family: Roboto;
    font-size: 12px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.29px;
    color: #000000;
`

const StyledTypographySubtotal = styled(Typography)`
    width: 296px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: #000000;
`

function HistoryCard() {
    return (
        <StyledCard>
            <StyledCardContent >
                <StyledTypographyRestaurantTitle>{props.restaurantName}</StyledTypographyRestaurantTitle>
                <StyledTypographyData>{props.period}</StyledTypographyData>
                <StyledTypographySubtotal>Subtotal R${props.value}</StyledTypographySubtotal>
            </StyledCardContent>
        </StyledCard>
    );
}

export default HistoryCard;