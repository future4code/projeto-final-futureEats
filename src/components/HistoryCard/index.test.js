import React from "react"
import { render } from "enzyme"
import HistoryCard from "."

describe("HistoryCard", () => {
    it("HistoryCard Snapshot", () => {
        const tree = render(<HistoryCard />)
        expect(tree).toMatchSnapshot()
    })
})