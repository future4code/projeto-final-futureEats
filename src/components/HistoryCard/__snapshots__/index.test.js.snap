// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryCard HistoryCard Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 sc-bdVaJa lmjXJd",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiCardContent-root-29 sc-bwzfXH dQsjzk",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-htpNat hpuEZk",
            },
            "children": Array [
              Object {
                "data": "Habbib's",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-bxivhb hCbXLp",
              },
              "children": Array [
                Object {
                  "data": "21 de Setembro de 2019",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-ifAKCX fSaSkM",
                },
                "children": Array [
                  Object {
                    "data": "Subtotal R$10.00",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-bxivhb hCbXLp",
            },
            "children": Array [
              Object {
                "data": "21 de Setembro de 2019",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-ifAKCX fSaSkM",
              },
              "children": Array [
                Object {
                  "data": "Subtotal R$10.00",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-htpNat hpuEZk",
              },
              "children": Array [
                Object {
                  "data": "Habbib's",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-ifAKCX fSaSkM",
            },
            "children": Array [
              Object {
                "data": "Subtotal R$10.00",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-bxivhb hCbXLp",
              },
              "children": Array [
                Object {
                  "data": "21 de Setembro de 2019",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "MuiTypography-root-30 MuiTypography-body1-39 sc-htpNat hpuEZk",
                },
                "children": Array [
                  Object {
                    "data": "Habbib's",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
